snooty/postprocess.py

```diff
@@ -2193,23 +2195,162 @@ def exit_page(self, fileid_stack: FileIdStack, page: Page) -> None:
+                    "selections",
+                    "at least one",
+                    node.span[0],
+                )
+            )
+            return
+
+        selections: Dict[str, str] = {}
+        for idx in range(len(selection_ids)):
+            selection_id = selection_ids[idx]
+            try:
+                composable_option = self.composable_tutorial.composable_options[idx]
+            except IndexError:
+                print("check index error caused by selection_id ", selection_id)
+                self.context.diagnostics[fileid_stack.current].append(
+                    InvalidChildCount(
+                        "selected-content",
+                        "selections",
+                        str(len(self.composable_tutorial.composable_options)),
+                        node.span[0],
+                    )
+                )
+                continue
+
+            composable_id = composable_option.get("value")
+            if not isinstance(composable_id, str):
+                self.context.diagnostics[fileid_stack.current].append(
+                    UnknownOptionId(
+                        "selections",
+                        str(composable_id) if composable_id is not None else "None",
+                        [
+                            spec_composable.id
+                            for spec_composable in self.spec.composables
+                        ],
+                        node.span[0],
+                    )
+                )
+                continue
+
+            spec_composable = next(
+                (
+                    spec_composable
+                    for spec_composable in self.spec.composables
+                    if spec_composable.id == composable_id
+                ),
+                None,
+            )
+            if not spec_composable:
+                self.context.diagnostics[fileid_stack.current].append(
+                    UnknownOptionId(
+                        "selections",
+                        composable_id,
+                        [
+                            spec_composable.id
+                            for spec_composable in self.spec.composables
+                        ],
+                        node.span[0],
+                    )
+                )
+                continue
+
+            selections[spec_composable.id] = selection_id
+
+            if not selection_id or selection_id == "None":
+                continue
+
+            # get option value+title from spec
+            # and insert into composable_option.selections
+            option_from_spec = next(
+                (
+                    spec_option
+                    for spec_option in spec_composable.options
+                    if spec_option.id == selection_id
+                ),
+                None,
+            )
+            if not option_from_spec:
+                continue
+
+            composable_option["selections"] = composable_option["selections"] or []
```

Comment:
Huh, is this basically the python alternative to `??=`? Like a nullish assignment?

Reply:
yup! same as [these lines moved](https://github.com/mongodb/snooty-parser/pull/670/files/4a659dec7bc7bed2ff77f4f02877102d1d874f54#diff-5e66e8539da7ed36bf30c7a27807d04eb3c8c9a0d78649afcdc4a90cb1871efeL1015)